pkg/scheduler/core/generic_scheduler.go

```diff
@@ -95,6 +95,7 @@ func (f *FitError) Error() string {
 type genericScheduler struct {
 	cache                    schedulercache.Cache
 	equivalenceCache         *equivalence.Cache
+	nodeCacheSnapshots       map[string]*equivalence.NodeCacheSnapshot
```

Comment:
I think this member should be named so it's clearly related to eCache. However, I am not convinced that we should have this.

The reason that we have `cachedNodeInfoMap` is so that we can avoid rebuilding a snapshot from scratch every time. In `UpdateNodeNameToInfoMap`, we check the generation ID and decide whether we need to DeepCopy the NodeInfo object. Since eCache snapshots are just generation IDs, the only thing you (potentially) avoid is map allocation. (might be worth benchmarking with and without...)

Reply:
I added this only to avoid map allocation. I'll do some benchmarks. By the way I'm trying to figure out is it possible to use sequence numbers to present predicate key, then we can use slices to store generations ids and predicate results also. In that case, performance will be much better.

Reply:
I think if the snapshot was a member of equivalenceCache, you didn't need to pass it separately here.